src/secret.js

```diff
@@ -20,30 +20,39 @@ const secretPath = path.normalize(__dirname + "/../secret");
 let secret;
 
 // Used by the benchmark/loadtest tool.
-const get = function() {
+const get = function(done) {
+  if (secret) {
+    return done(null, done);
+  }
+  fs.readFile(secretPath, "utf-8", (err, contents) =>{
+    if (err) {
+      logging.error(`FATAL ERROR (${err}): You must create a file:`);
+      logging.error('    ' + secretPath);
+      logging.error('Its contents should be the secret-string at');
+      logging.error('    https://phabricator.khanacademy.org/K121');
+      return done(err);
+    }
+
+    var secret = contents.trim();
     if (!secret) {
-        try {
-            secret = fs.readFileSync(secretPath, "utf-8").trim();
-            if (!secret) {     // empty file?
-                throw new Error('secret file is empty!');
-            }
-        } catch (err) {
-            logging.error(`FATAL ERROR (${err}): You must create a file:`);
-            logging.error('    ' + secretPath);
-            logging.error('Its contents should be the secret-string at');
-            logging.error('    https://phabricator.khanacademy.org/K121');
-            throw err;
-        }
+        return done(new Error('secret file is empty!'));
     }
-    return secret;
+
+    return done(secret);
```

Comment:
Yeah, this was another thing I noticed on my second run through the code. I've added a separate test for the `secret` module so we can catch this stuff in the future.